crates/blockifier/src/state/visited_pcs.rs

```diff
@@ -0,0 +1,94 @@
+use std::collections::hash_map::{Entry, IntoIter, Iter};
+use std::collections::{HashMap, HashSet};
+use std::fmt::Debug;
+
+use starknet_api::core::ClassHash;
+
+use super::state_api::State;
+
+/// This trait is used in `CachedState` to record visited pcs of an entry point call.
```

Comment:
Missing docs. You should at least talk about the expected requirements on implementations, and the fact that the trait is not intended to be `dyn`-safe.